src/main/java/org/junit/rules/Timeout.java

```diff
@@ -143,16 +143,17 @@ protected final boolean getLookingForStuckThread() {
      * @since 4.12
      */
     protected Statement createFailOnTimeoutStatement(
-            Statement statement) throws Exception {
+            Statement statement, Description description) throws Exception {
```

Comment:
> We can mark the old method as `@Deprecated`, and add a new, private method with a signature that takes a `Description`. The old method would work as before, not providing a `Description` to the `FailOnTimeout`--thus threads created via the old method would be given generic names like `"Time-limited test"`.

Oops, no we can't--`Timeout#apply` calls `Timeout#createFailOnTimeoutStatement`. To keep both signatures of that method around we'd have to provide a hook for users to configure the behavior of `Timeout#apply`.

I'm in favor of making the method private, but how does this jive with the [deprecation policy](https://github.com/junit-team/junit/wiki/Deprecation-Policy)?